src/mfxloader.cpp

```diff
@@ -452,7 +452,21 @@ mfxStatus MFXVideoUSER_Load(mfxSession session, const mfxPluginUID *uid, mfxU32
       if (MFX::g_GlobalCtx.m_plugins.empty()) {
         // Parsing plugin configuration file and loading information of
         // _all_ plugins registered on the system.
+#if defined(__linux__)
+        // Hardcode common plugins paths for most linux distros in case of
+        // the legacy mfxdispatcher is configured to static build.
+        const char *plugins_paths[3] = { "/usr/share/mfx/plugins.cfg",
+                                         "/usr/local/share/mfx/plugins.cfg",
+                                         "/opt/intel/mediasdk/plugins/plugins.cfg" };
+        for (int i = 0; i < sizeof(plugins_paths) / sizeof(plugins_paths[0]); i++) {
+          if (fopen(plugins_paths[i], "r") != NULL) {
```

Comment:
Doesn't this leave an open file descriptor dangling around?

Reply:
Fixed.